File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        i = j = 0
        n = len(s)
        m = len(t)
        while j < m:
            if i == n:
                return m - j
            if s[i]==t[j]:
                j = j + 1
            i = i + 1
        return 0

File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {

        List <List <Integer>> result = new ArrayList<>();


        return subset(nums,0,new ArrayList<Integer>(), result);
    }

    public List<List<Integer>> subset (int nums[], int i, List <Integer> sub, List<List<Integer>> result)
    {
        ArrayList<Integer> newSub = new ArrayList<>(sub);
        newSub.add(nums[i]);
        if(i==nums.length - 1)
        {
            result.add(sub);
            result.add(newSub);
            return result;
        }
        i++;
        subset(nums, i, sub, result);
        subset(nums, i, newSub, result);

        return result;
    }
}
